endpoints/openrtb2/video_auction.go

```diff
@@ -79,7 +81,27 @@ func (deps *endpointDeps) VideoAuctionEndpoint(w http.ResponseWriter, r *http.Re
 		CookieFlag:    pbsmetrics.CookieFlagUnknown,
 		RequestStatus: pbsmetrics.RequestStatusOK,
 	}
+
+	debugQuery := r.URL.Query().Get("debug")
+	enableDebug := debugQuery == "true"
+	debugString := ""
+	errorCacheID := ""
+
 	defer func() {
+		if enableDebug && vo.VideoResponse == nil {
+			cache := deps.ex.GetCache()
+			debugString = fmt.Sprintf("<!--\n%s\n-->", debugString)
+			data, _ := json.Marshal(debugString)
+			toCache := []prebid_cache_client.Cacheable{
+				{
+					Type:       prebid_cache_client.TypeXML,
+					Data:       data,
+					TTLSeconds: int64(deps.cfg.CacheURL.DefaultTTLs.Video),
+					Key:        fmt.Sprintf("log_%s", errorCacheID),
```

Comment:
Is errorCacheID guaranteed to be set to something here? In other words we're sure handleError is called in every case where vo.VideoResponse doesn't get set?

Actually this gives me an idea about how we might be able to avoid having errorCacheID be returned by the handleError() function but still have it in the response, remind me tomorrow and we can discuss this either in person or via slack.

Reply:
Ah I should be checking if it's set to something here. I could use a string pointer instead and modify in the `handleError`. Then I can check if it is empty. Changing this is quick though so we can discuss it tomorrow.

Reply:
I'm good with the cache put here, since it will only happen if the auction didn't write to the cache ensuring only a single cache put per request.

Please add checks in this code to defend against nils and errors. Specifically, ensure that `cache` is not nil before calling `PutJson`, do not ignore the error from json.Marshal. The TTLSeconds conversion also looks a bit risky to me, is it guaranteed here to succeed? 

For the key formatting, it might be better to use simple string concatenation in this instance.

Reply:
Added checks around the caching to ensure that it is safe to make the call. Also updated key formatting.

For the TTL, should I set a default value and then check if the config has something, or if the config has no TTL value, ignore the caching?